domain/lib/src/extension/string_extension.dart

```diff
@@ -43,6 +43,8 @@ extension StringExtension on String {
 
   String toMiddleEllipsis() {
     final subLength = length - split('.').last.length;
-    return '...' + substring(subLength - 3, subLength) + split('.').last;
```

Comment:
Some file may have no extension, so I think we should check more cases, like this:

```dart
if(contains('.')) {
	final subLength = length - split('.').last.length;
	return subLength >= 3
	  ? '...' + substring(subLength - 3, subLength) + split('.').last
	  : this;
} else {
  	return length >= 3 ? '...' + substring(length - 3, length) : this;
}
```

Or to use package https://pub.dev/packages/path to check extension exists